src/lib/features/feature-toggle/feature-toggle-service.ts

```diff
@@ -1587,8 +1587,22 @@ class FeatureToggleService {
         );
     }
 
-    async validateArchiveToggles(featureNames: string[]): Promise<string[]> {
-        return this.dependentFeaturesReadModel.getOrphanParents(featureNames);
+    async validateArchiveToggles(featureNames: string[]): Promise<{
+        hasDeletedDependencies: boolean;
+        parentsWithChildFeatures: string[];
+    }> {
+        const hasDeletedDependencies =
+            await this.dependentFeaturesReadModel.haveDependencies(
+                featureNames,
+            );
+        const parentsWithChildFeatures =
+            await this.dependentFeaturesReadModel.getOrphanParents(
+                featureNames,
+            );
+        return {
+            hasDeletedDependencies,
```

Comment:
added this to our archive validation. It now indicates if any dependencies will be removed